core/blockchain_test.go

```diff
@@ -42,6 +44,79 @@ var (
 	forkSeed      = 2
 )
 
+func TestUptimeSingle(t *testing.T) {
+	var uptimes []istanbul.Uptime
+	uptimes = updateUptime(uptimes, 212, big.NewInt(7), 3, 2, 211)
+	// the first 2 uptime updates do not get scored since they're within the
+	// first window after the epoch block
+	expected := []istanbul.Uptime{
+		{
+			Score:           0,
+			LastSignedBlock: 211,
+		},
+		{
+			Score:           0,
+			LastSignedBlock: 211,
+		},
+		{
+			Score:           0,
+			LastSignedBlock: 211,
+		},
+		// plus 2 dummies due to the *1.5
```

Comment:
I do not have an immediate idea on how we can remove these dummies, although they're just going to be empty so don't think they're much harm.

Reply:
I think the way around this is to make the datastructure be an array whose length is equal to the highest order bit that we've seen in the bitmap thus far. But I don't think we should be in a hurry to fix this